torch/csrc/jit/register_special_ops.cpp

```diff
@@ -112,7 +111,7 @@ void storeLastDimension<bool>(
   auto seq_size = obj.size();
   checkSequenceSize(n, dim, seq_size);
   for (int64_t i = 0; i < n; i++) {
-    *(uint8_t*)data = static_cast<uint8_t>(obj[i]);
+    *(bool*)data = static_cast<bool>(obj[i]);
```

Comment:
I think you need to change   the line here https://github.com/pytorch/pytorch/blob/6fb58f7efd2b615e4103029f6be5244a162e46e2/torch/csrc/jit/register_special_ops.cpp#L293 

from `at::kByte` to the new Dtype